### PR TITLE
Api support read only

### PR DESCRIPTION
**Pull request by helto4real**
## Proposed change

This adds basic API to NetDaemon. First MVP will be a read-only information about the applications and the settings. Future releases will have possibility to interact. This interface is also built on http GET. Future might implement websockets. 

## API
| path | description |
| ----- | ------------ |
| /api/apps | returns a list of the apps and information about them |
| /api/settings | returns the settings. Token is always empty for now |

api accepts localhost only on port 5000 for now. 

## noteworthy 

This release uses ASP.NET instead of general host for NetDaemon. 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -26,9 +26,12 @@
 	},
 	"postCreateCommand": "dotnet restore && .devcontainer/install_prettyprompt.sh",
 	// Uncomment the next line if you want to publish any ports.
-	"appPort": [
-		5001
+	"forwardPorts": [
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:29]: It's not really needed to publish this.
If you want to test it you can just do `curl localhost:5000/api/apps` inside the container

File: src/DaemonRunner/DaemonRunner/Service/RunnerService.cs [hunk 2]
```diff
@@ -88,70 +92,62 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
...
-                                            _logger.LogWarning($"Home assistant is unavailable, retrying in {ReconnectInterval / 1000} seconds...");
-                                        }
+                                        (loadedDaemonApps, alc) = DaemonCompiler.GetDaemonApps(sourceFolder!, _logger);
                                     }
-                                    catch (TaskCanceledException)
+
+                                    if (loadedDaemonApps is null || !loadedDaemonApps.Any())
                                     {
-                                        _logger.LogInformation("Canceling NetDaemon service...");
+                                        _logger.LogWarning("No .cs files files found, please add files to [netdaemonfolder]/apps");
```
Comment (ludeeus) [inline src/DaemonRunner/DaemonRunner/Service/RunnerService.cs:123]:
```suggestion
                                        _logger.LogWarning($"No .cs files files found, please add files to {sourceFolder}/apps");
```

Comment (helto4real) [reply src/DaemonRunner/DaemonRunner/Service/RunnerService.cs:123]: Thanks for reminding me about this one!

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (helto4real) [reply .devcontainer/devcontainer.json:29]: Yea true, I did test it like that but wanted to be sure ... I will remove it

**Comment by helto4real** [comment]
it is coveralls that failing the tests so it works

**Pull request merged**